arangod/Aql/OptimizerRules.cpp

```diff
@@ -7471,6 +7470,29 @@ void arangodb::aql::geoIndexRule(Optimizer* opt,
     LimitNode* limit = nullptr;
     bool canUseSortLimit = true;
 
+    bool mustRespectIdxHint = false;
+    if (plan->hasForcedIndexHints()) {
+      auto enumerateColNode =
+          ExecutionNode::castTo<EnumerateCollectionNode const*>(node);
+      auto indexes = enumerateColNode->collection()->indexes();
+      auto idxNames = enumerateColNode->hint().hint();
+      for (auto const& idxName : idxNames) {
+        for (std::shared_ptr<Index> idx : indexes) {
```

Comment:
copies each shared_ptr...
```suggestion
        for (std::shared_ptr<Index> const& idx : indexes) {
```